### PR TITLE
python3Packages.dnspython: 2.1.0 -> 2.2.0

### PR DESCRIPTION
**Pull request by gador**
Update of dnspython

###### Motivation for this change
needed for pgadmin4 init
https://github.com/NixOS/nixpkgs/pull/154764

This version includes a bugfix for missing /etc/protocols files and is needed for building pgadmin4

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage) 
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4250,6 +4250,16 @@
...
+  gador = {
+    email = "florian.brandes@posteo.de";
+    github = "gador";
+    githubId = 1883533;
+    name = "Florian Brandes";
+    keys = [{
+      longkeyid = "rsa4096/0xBBB3E40E53797FD9";
+      fingerprint = "0200 3EF8 8D2B CF2D 8F00  FFDC BBB3 E40E 5379 7FD9";
+    }];
+  };
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:4262]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/development/python-modules/dnspython/default.nix [hunk 2]
```diff
@@ -2,26 +2,32 @@
 , buildPythonPackage
 , fetchPypi
 , pythonOlder
+, setuptools-scm
 }:
 
 buildPythonPackage rec {
   pname = "dnspython";
-  version = "2.1.0";
+  version = "2.2.0rc1";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dnspython/default.nix:10]: I think we want to update with this update until the full release.

**Comment by gador** [comment]
evaluation fails due to missing maintainer info #155411

File: pkgs/development/python-modules/dnspython/default.nix (hunk 2 above)
Comment (gador) [reply pkgs/development/python-modules/dnspython/default.nix:10]: I [asked](https://github.com/rthalley/dnspython/issues/742#issuecomment-1014892484) and we probably won't have to wait long.

Comment (gador) [reply pkgs/development/python-modules/dnspython/default.nix]: new version is out. I also added tests, since they do not touch the network anymore

File: pkgs/development/python-modules/dnspython/default.nix [hunk 3]
```diff
@@ -2,26 +2,40 @@
...
   };
 
-  # needs networking for some tests
-  doCheck = false;
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  disabledTests = [
+    # dns.exception.SyntaxError: protocol not found
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dnspython/default.nix:25]: If ``/etc/protocols`` is missing, than that can be fixed like this https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/eventlet/default.nix#L46-L48 (minus the ``/etc/resolve`` part.

Comment (gador) [reply pkgs/development/python-modules/dnspython/default.nix:25]: unfortunately, the same test then fails with "dns.exception.SyntaxError: service/proto not found"
I would just disable this particular test
